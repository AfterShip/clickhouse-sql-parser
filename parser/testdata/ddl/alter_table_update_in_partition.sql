ALTER TABLE test.users UPDATE status = 'active' IN PARTITION '2024-01' WHERE id > 100;
