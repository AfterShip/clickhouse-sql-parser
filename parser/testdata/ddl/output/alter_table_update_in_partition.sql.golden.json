[
  {
    "AlterPos": 0,
    "StatementEnd": 85,
    "TableIdentifier": {
      "Database": {
        "Name": "test",
        "QuoteType": 1,
        "NamePos": 12,
        "NameEnd": 16
      },
      "Table": {
        "Name": "users",
        "QuoteType": 1,
        "NamePos": 17,
        "NameEnd": 22
      }
    },
    "OnCluster": null,
    "AlterExprs": [
      {
        "UpdatePos": 23,
        "StatementEnd": 85,
        "Assignments": [
          {
            "AssignmentPos": 30,
            "Column": {
              "Ident": {
                "Name": "status",
                "QuoteType": 1,
                "NamePos": 30,
                "NameEnd": 36
              },
              "DotIdent": null
            },
            "Expr": {
              "LiteralPos": 40,
              "LiteralEnd": 46,
              "Literal": "active"
            }
          }
        ],
        "WhereClause": {
          "LeftExpr": {
            "Name": "id",
            "QuoteType": 1,
            "NamePos": 77,
            "NameEnd": 79
          },
          "Operation": "\u003e",
          "RightExpr": {
            "NumPos": 82,
            "NumEnd": 85,
            "Literal": "100",
            "Base": 10
          },
          "HasGlobal": false,
          "HasNot": false
        },
        "InPartition": {
          "PartitionPos": 51,
          "Expr": {
            "LiteralPos": 62,
            "LiteralEnd": 69,
            "Literal": "2024-01"
          },
          "ID": null,
          "All": false
        }
      }
    ]
  }
]